shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/reconcilers/BridgeExecutorDeploymentReconciler.java

```diff
@@ -0,0 +1,74 @@
+package com.redhat.service.smartevents.shard.operator.reconcilers;
+
+import com.redhat.service.smartevents.shard.operator.DeltaProcessorService;
+import com.redhat.service.smartevents.shard.operator.comparators.Comparator;
+import com.redhat.service.smartevents.shard.operator.comparators.DeploymentComparator;
+import com.redhat.service.smartevents.shard.operator.exceptions.ReconciliationException;
+import com.redhat.service.smartevents.shard.operator.resources.BridgeExecutor;
+import com.redhat.service.smartevents.shard.operator.resources.BridgeExecutorStatus;
+import com.redhat.service.smartevents.shard.operator.services.BridgeExecutorDeploymentService;
+import com.redhat.service.smartevents.shard.operator.services.StatusService;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.List;
+
+@ApplicationScoped
+public class BridgeExecutorDeploymentReconciler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BridgeExecutorDeploymentReconciler.class);
+
+    @Inject
+    DeltaProcessorService deltaProcessorService;
+
+    @Inject
+    BridgeExecutorDeploymentService bridgeExecutorDeploymentService;
+
+    @Inject
+    StatusService statusService;
+
+    public void reconcile(BridgeExecutor bridgeExecutor){
+
+        try{
+            List<Deployment> requestResource = createRequiredResources(bridgeExecutor);
+
+            List<Deployment> deployedResources = fetchDeployedResources(bridgeExecutor);
+
+            processDelta(requestResource, deployedResources);
+
+            validateDeploymentIsReady(bridgeExecutor);
```

Comment:
@vaibhavjainwiz I have been considering this _solution_ for premature success notifications.

Unfortunately this still has the potential to incorrectly notify intermittent failure statuses to the `manager` when the User updates a failed resource. Consider the following: 

1. A `Deployment` has its Conditions in a _failed_ state for whatever reason; and the Processor is marked as failed to the User.
2. The User updates the Processor definition and the Operator's reconcile loop is eventually re-entered.
3. When L#41 is invoked you _signal_ to k8s to update the `Deployment`. This update will be handled by another [Operator](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/deployment_controller.go) somewhere at some point in time.
4. You immediately retrieve the `Deployment`'s status on L#43 which _could be before_ the Operator has had sufficient time to attempt to re-provision the revised `Deployment`.
5. This could lead to the `validateDeploymentIsReady(..)` throwing `ReconciliationException` and signalling to the `manager` that the Update has failed.
6. When the reconcile loop runs again the `Deployment` Operator may have re-provisioned the resource and L#43 retives the updated `Deployment` status; no `ReconciliationException` is thrown and the `manager` signalled that everything was successful.

This behaviour is no different to v1 but something we was hoping to improve on with v2.

I think you ideally do not want to retrieve the `Deployment` status until the next reconcile loop iteration triggered by the `Deployment`'s `EventSource`. Reading the "[Patterns and Best Practices](https://javaoperatorsdk.io/docs/patterns-best-practices#eventsources-and-caching)" suggests both v1 and v2 implementations are flawed in that they should not be "...Reading the actual state of a resource from the Kubernetes API Server directly all the time..." but should respond to the `EventSource` triggers.

With the code split you have proposed; it should be reasonably simple to have _your_ reconcilers respond only to `EventSource` triggers and not on every reconcile loop. It may be something worth investigating and considering.